### PR TITLE
docs: dependabot security suggestion

### PR DESCRIPTION
**Pull request by ulysses4ever**
Cf. https://github.com/haskell/cabal/security/dependabot/3

---
Please include the following checklist in your PR:

* [ ] Patches conform to the [coding conventions](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#conventions).
* [ ] Any changes that could be relevant to users [have been recorded in the changelog](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#changelog).
* [ ] The documentation has been updated, if necessary.
* [ ] Include [manual QA notes](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#qa-notes) if your PR relates to cabal-install.

Bonus points for added automated tests!

### CODE REVIEW

File: doc/requirements.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #
-# This file is autogenerated by pip-compile with Python 3.10
+# This file is autogenerated by pip-compile with Python 3.8
```
Comment (ffaf1) [inline doc/requirements.txt:2]: I am not sure why the downgrade to 3.8, but I guess it doesn’t break anything either.

**Review by ffaf1 — APPROVED** [review]
very good

**Review by Kleidukos — APPROVED** [review]

File: doc/requirements.txt (hunk 1 above)
Comment (ulysses4ever) [reply doc/requirements.txt:2]: Yes, I was also annoyed by it. The issue is: the machine I was doing it on has only 3.8, no way around it. The files is generated by Python and puts its version there. I could manually edit it back to 3.10 but that would be a lie. I thought that I rather avoid lying here at the expense of a bigger diff. Functionally, there's probably no difference...

Comment (ulysses4ever) [reply doc/requirements.txt]: All right, god bless Docker, I did it with 3.10...

**Comment by ulysses4ever** [comment]
I think I'll add something to `doc/README.md` 'cause it's excruciatingly painful to remember what to do every time dependabot hit us up.

**Comment by ulysses4ever** [comment]
I added the docs but it looks like RTD is not happy with too new dependencies now. Seems it uses Python 3.7...

**Comment by ulysses4ever** [comment]
All right, it was us who pinned Python 3.7 in `.readthedocs.yaml`. Bumped to 3.10...

**Comment by ulysses4ever** [comment]
RTD supports Python 3.8 max :sob: Builds now.

**Pull request merged**